### ISSUE TITLE
access_denied_listener is using TwigBundle\Controller\ExceptionController

### ISSUE THREAD

**Issue by estahn** [FriendsOfSymfony/FOSRestBundle#1356]
With `access_denied_listener` set to `true` (for json) it will use the `TwigBundle\Controller\ExceptionController` even though the configuration `templating_formats` for `json` is set to `false`.

Throwing an exception usually works as excepted with the following error message as example:

```
throw new HttpException(400, "foo"); -> {"code":400,"message":"foo"}
```

Now, using `@Security("has_role('ROLE_FOOBAR')")` it will throw an AccessDenied exception which will go through `TwigBundle\Controller\ExceptionController` resulting in the following output:

```{"error":{"code":403,"message":"Forbidden","exception":[{"message":"You do not have the necessary permissions","class":"Symfony\\Component\\HttpKernel\\Exception\\AccessDeniedHttpException"...```

I could only find the following options to solve it:

1. Disable `access_denied_listener`
2. Create twig template for json response

In regards to _2._, create a file `app/Resources/TwigBundle/views/Exception/exception.json.twig` with the following content:
```{{ { 'code': status_code, 'message': status_text, 'exception': exception.toarray }|json_encode|raw }}```

I would like to know if this is expected behaviour or something is not correctly configured. I would assume everything should go through the same exception controller.

I had a hard time finding the actual issue or how to resolve it. In case this is expected behaviour this ticket can be closed and provide some guidance.

**Comment by lsmith77** [comment]
An uncaught exception should always go to the configured exception handler .. if you enable the FOSRestBundle exception handler, than its also to be expected that it handles this exception. But if you do not enable the FOSRestBundle exception controller .. it will use the one from Twig if that is enabled.

So the solution is enable the FOSRestBundle exception controller.

**Comment by Ener-Getick** [comment]
BTW @lsmith77 why does the AccessDeniedListener extend the twig ExceptionListener ?
It could only change the event exception and let the other listeners configured render the response right ?

**Comment by lsmith77** [comment]
not sure ..

**Comment by estahn** [comment]
@lsmith77 You're saying i haven't configured the FOSRestBundle exception handler and need to do this as well? I will give it a try and report back. Thanks for the hint.

**Comment by Ener-Getick** [comment]
https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1363 should fix your issue

**Comment by Ener-Getick** [comment]
Fixed by #1363 (master branch).
